src/Compositor.cpp

```diff
@@ -996,11 +996,75 @@ void CCompositor::updateWindowBorderColor(CWindow* pWindow) {
         pWindow->m_cRealBorderColor = CColor(pWindow == m_pLastWindow ? *ACTIVECOL : *INACTIVECOL);
 }
 
-void CCompositor::moveWindowToWorkspace(CWindow* pWindow, const std::string& work) {
-    m_pLastWindow = pWindow;
-    g_pKeybindManager->moveActiveToWorkspace(work);
+void CCompositor::moveWindowToWorkspace(CWindow* pWindow, const int pWorkspaceID) {
+
+    if (!g_pCompositor->windowValidMapped(pWindow))
+        return;
+
+    const auto OLDWORKSPACE = g_pCompositor->getWorkspaceByID(pWindow->m_iWorkspaceID);
+
+    if (pWorkspaceID == pWindow->m_iWorkspaceID) {
+        Debug::log(LOG, "Not moving to workspace because it didn't change.");
+        return;
+    }
+
+    g_pLayoutManager->getCurrentLayout()->onWindowRemoved(pWindow);
```

Comment:
we should do it after the return checks to avoid just yeeting the window to narnia